specreduce/extract.py

```diff
@@ -549,18 +730,32 @@ def __call__(self, image=None, trace_object=None,
                 norms.append(np.nan)
                 continue
 
-            # else, set compound model's mean to column's matching trace value
-            fit_ext_kernel.mean_0 = mean_init_guess[col_pix]
-
-            # NOTE: support for variable FWHMs forthcoming and would be here
-
-            # fit compound model to column
-            fitted_col = fit_ext_kernel(xd_pixels)
-
-            # save result and normalization
-            kernel_vals.append(fitted_col)
-            norms.append(fit_ext_kernel.amplitude_0
-                         * fit_ext_kernel.stddev_0 * np.sqrt(2*np.pi))
+            if self.spatial_profile == 'gaussian':
+
+                # else, set compound model's mean to column's matching trace value
+                # again, this is probably not necessary
+                if bkgrd_prof is not None:  # attr names will be diff. if not compound
+                    fit_ext_kernel.mean_0 = mean_init_guess[col_pix]
+                else:
+                    fit_ext_kernel.mean = mean_init_guess[col_pix]
+
+                # evaluate fit model (with shifted mean, based on trace)
+                fitted_col = fit_ext_kernel(xd_pixels)
+
+                # save result and normalization
+                # this doesn't need to be in this loop, address later
```

Comment:
is there an issue tracking this?  If not, should we create one so we don't forget?

Reply:
no, i plan to address this when i address masking. right now it makes sense for it to be in the loop, even though the area under the gaussian doesn't change when the mean shifts (and, we don't need to shift the mean at all since align_along_trace is accounting for this), but it does append NaN to the output array if the column has any masked values which is how masks are treated for now.